src/vea/vea_api.c

```diff
@@ -53,6 +56,29 @@ vea_upgrade(struct vea_space_info *vsi, struct umem_instance *umem,
 	if (rc != 0)
 		return rc;
 
+	/*
+	 * bitmap tree reused vec tree which was created with
+	 * BTR_FEAT_DIRECT_KEY, recreate tree with BTR_FEAT_UINT_KEY
+	 */
+	if (daos_handle_is_valid(vsi->vsi_md_bitmap_btr)) {
+		dbtree_destroy(vsi->vsi_md_bitmap_btr, NULL);
```

Comment:
This looks dangerous to me, is it possible that vea_upgrade() is called on an upgraded pool?

Reply:
I agreed, i improved vea_upgrade a bit to check vsd_compat before upgrading,
if feature bits existed, we don't do this dangerous operations.

Reply:
We might also add an assert here like:
D_ASSERT(dbtree_is_empty(xxx))